index.d.ts

```diff
@@ -131,31 +138,167 @@ declare module 'warframe-items' {
+        category: 'Mods' | 'Misc';
+        type: 'Riven Mod' | `${WeaponType} Riven Mod`;
+        availableChallenges?: AvailableChallenge[];
+        upgradeEntries?: UpgradeEntry[];
+        polarity?: Polarity;
+    }
+    interface SingleLevelMod extends Omit<Mod, 'levelStats'> {}
+    type ModType = 
+        'Railjack' |
+        'Necramech' |
+        'Warframe' |
+        'Secondary' |
+        'Melee' |
+        'Companion' |
+        'Primary' |
+        'K-Drive' |
+        'Riven' |
+        'Archwing' |
+        'Arch-Melee' |
+        'Arch-Gun' |
+        'Shotgun' |
+        'Creature' |
+        'Stance' |
+        'Parazon' |
+        'Transmutation' |
+        'Peculiar';
+    interface Mod extends MinimalItem, WikiaItem, Droppable {
+        baseDrain: number;
+        category: 'Mods';
+        compatName?: string;
+        fusionLimit: number;
+        isAugment?: boolean;
+        isPrime: false;
+        levelStats?: LevelStat[];
+        polarity: Polarity;
+        transmutable?: boolean;
+        /** @deprecated probably replaced by Mod#isExilus */
+        isUtility?: boolean;
+        modSet?: UniqueName;
+        isExilus?: boolean;
+        modSetValues?: number[];
+        type: `${ModType} Mod`;
+    }
+    interface PrimeMod extends Omit<Mod, 'isPrime'> { isPrime: true; }
+    interface Warframe extends Equippable, Buildable, WikiaItem, BaseItem {
+        category: 'Warframes' | 'Archwing' | 'Pets' | 'Sentinels';
+        health: number;
+        shield: number;
+        armor: number;
+        stamina: number;
+        power: number;
+        sprintSpeed?: number;
+        abilities?: Ability[];
+        aura?: Polarity;
+        sex?: Sex;
+        sprint?: number;
+        passiveDescription?: string;
+        maxLevelCap?: number;
+        exalted?: Array<UniqueName>;
+    }
+    interface SolNode extends MinimalItem {
+        category: 'Node';
+        systemIndex: number;
+        systemName: SystemName;
+        nodeType: number;
+        missionIndex: number;
+        factionIndex?: number;
+        minEnemyLevel: number;
+        maxEnemyLevel: number;
+        masteryReq: number;
+        tradable: false;
+        type: 'Node';
+        drops?: Drop[];
+    }
+    interface Extractor extends MinimalItem, Buildable {
+        category: 'Misc';
+        binCount: number;
+        binCapacity: number;
+        fillRate: number;
+        durability: number;
+        repairRate: number;
+        capacityMultiplier?: number[];
+        specialities: string[];
+    }
+    interface Skin extends MinimalItem, Equippable {
+        category: 'Skins';
+        excludeFromCodex: true;
+    }
+    interface Relic extends MinimalItem {
+        category: 'Relics';
+        type: 'Relic';
+        locations: Array<TitaniaRelicLocation>;
+        rewards: Array<TitaniaRelicReward>;
+        vaulted: boolean;
+        vaultDate: string;
+        marketInfo: TitaniaWFMInfo | null;
```

Comment:
Is it necessary to have an union with null here? As every relic is tradable and then would have a link to warframeMarket, but considering the case of a new releases item it takes some time for it to be in warframe market.

Reply:
That seems like you answered your own question....